arnold/__init__.py

```diff
@@ -1,164 +1,242 @@
-    else:
-        model.delete().where(
-            model.migration == migration
-        ).execute()
 
+def up(args):
+    Terminator(args).perform_migrations('up')
 
-def _perform_migrations(direction, model, **kwargs):
-    """
-    Find the migration if it is passed in and call the up or down method as
-    required. If no migration is passed, loop through list and find
-    the migrations that need to be run.
-    """
-    migration = kwargs.get("migration")
-    files = os.listdir(kwargs.get("directory"))
-    filenames = _retreive_filenames(files)
 
-    if direction == "down":
-        filenames.reverse()
+def down(args):
+    Terminator(args).perform_migrations('down')
 
-    if len(filenames) <= 0:
-        return True
 
-    if migration:
-        if migration not in filenames:
-            raise MigrationNotFoundException
-        _perform_single_migration(direction, model, **kwargs)
-        return True
+def status(args):
+    latest_migration = Terminator(args).get_latest_migration()
+    if latest_migration:
+        print("Migration {0} run at {1}.".format(
+            colored(latest_migration.migration, "blue"),
+            colored(latest_migration.applied_on, "green"),
+        ))
+    else:
+        print("No migrations currently run.")
 
-    for f in filenames:
-        _perform_single_migration(direction, model, migration=f, **kwargs)
+        
+def init(args):
+    os.makedirs('{0}/migrations'.format(args.folder))
+    open('{0}/__init__.py'.format(args.folder), 'a').close()
+    open('{0}/migrations/__init__.py'.format(args.folder), 'a').close()
     return True
+         
 
+def parse_args(args):
+    sys.path.insert(0, os.getcwd())
+    parser = argparse.ArgumentParser(description='Migrations. Down. Up.')
+    subparsers = parser.add_subparsers(help='sub-command help')
 
-def main(direction="up", **kwargs):
-    """The main method, handle exceptions and start migrations"""
-    # Pop ignored and db, they aren't required later
-    ignored = kwargs.pop("ignored", None)
-    try:
-        db = kwargs.pop("database")
-    except KeyError:
-        raise DBAttrNotFound
-    # Get directory and module, we need them later
-    directory = kwargs.get("directory")
-    migration_module = kwargs.get("migration_module")
+    init_cmd = subparsers.add_parser('init', help='Create the config folder.')
+    init_cmd.set_defaults(func=init)
 
-    if direction not in ["up", "down"]:
-        raise ArgumentException
-
-    if ignored:
-        IGNORED_FILES.extend(ignored)
-
-    if db:
-        model = Migration
-        model._meta.database = db
+    if args[0] == 'init':
+        init_cmd.add_argument(
+            '--folder', default='arnold_config', help='The folder to create.'
+        )
     else:
-        raise DBAttrNotFound
+        parser.add_argument(
+            '--folder', dest='folder', default='arnold_config',
+            help='The folder that contains arnold files.'
+        )    
```

Comment:
Why not make the `folder` option available at the very early stage? We're using arnold to install migrations from external packages, this will not work if it is not available.